src/cls/rbd/cls_rbd.cc

```diff
@@ -3862,23 +3864,35 @@ int mirror_mode_get(cls_method_context_t hctx, bufferlist *in,
     return r;
   }
 
+  std::string data_pool_name;
+  r = read_key(hctx, mirror::DATA_POOL_NAME, &data_pool_name);
+  if (r < 0 && r != -ENOENT) {
+    return r;
+  }
+
   ::encode(mirror_mode_decode, *out);
+  ::encode(data_pool_name, *out);
   return 0;
 }
 
 /**
  * Input:
  * @param mirror_mode (cls::rbd::MirrorMode) (uint32_t)
+ * @param data_pool_name (std::string)
  *
  * Output:
  * @returns 0 on success, negative error code on failure
  */
 int mirror_mode_set(cls_method_context_t hctx, bufferlist *in,
                     bufferlist *out) {
   uint32_t mirror_mode_decode;
+  std::string data_pool_name;
   try {
     bufferlist::iterator bl_it = in->begin();
     ::decode(mirror_mode_decode, bl_it);
+    if (!bl_it.end()) {
```

Comment:
I believe this will keep backward compatibility for mirroring that was configured before the data-pool option was added. It would probably be nicer to add a versioned structure for this config, but I'm not sure if it's worth it.